crates/meroctl/src/version.rs

```diff
@@ -0,0 +1,50 @@
+use rand::Rng;
+use reqwest::Client;
+use semver::Version;
+use serde::Deserialize;
+
+const CURRENT_VERSION: &str = env!("CARGO_PKG_VERSION");
+
+pub async fn check_for_update() {
+    let mut rng = rand::thread_rng();
+    let n: u8 = rng.gen();
+    if n != 11 {
+        return;
+    }
+    let url = "https://api.github.com/repos/calimero-network/core/releases/latest";
+
+    let client = Client::new();
+    let response = match client
+        .get(url)
+        .header("User-Agent", "meroctl-version-check")
+        .send()
+        .await
+    {
+        Ok(resp) => resp,
+        Err(_) => return,
+    };
+
+    #[derive(Deserialize)]
+    struct Release {
+        tag_name: String,
+    }
+
+    let release: Release = match response.json().await {
+        Ok(json) => json,
+        Err(_) => return,
+    };
+
+    let latest = match Version::parse(&release.tag_name) {
+        Ok(v) => v,
+        Err(_) => return,
+    };
+
+    let current = match Version::parse(CURRENT_VERSION) {
+        Ok(v) => v,
+        Err(_) => return,
+    };
+
+    if latest > current {
+        println!("A new version {latest} for meroctl is available.");
```

Comment:
the old version was pretty nice

![CleanShot 2025-05-09 at 10 26 04@2x](https://github.com/user-attachments/assets/5dbe80b0-503b-4eee-a5b5-fdee516b1ae4)